frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/VariantForm/VariantForm.tsx

```diff
@@ -195,6 +197,16 @@ export const VariantForm = ({
 
     const [errors, setErrors] = useState<IVariantFormErrors>({});
 
+    const shouldAddTypeNumberToOptions = Boolean(
+        uiConfig.flags.variantTypeNumber
+    );
+
+    useEffect(() => {
```

Comment:
I'm assuming it's because the flag hasn't loaded yet, correct?

Reply:
payloadOptions is defined before the component - we need to load the flag and determine whether to add the option to the list